HW1/P3/P3.py

```diff
@@ -0,0 +1,24 @@
+import pyspark
+
+if __name__ == '__main__':
+  sc = pyspark.SparkContext(appName='YK-P3')
+  wlist = sc.textFile('words.txt')
+
+  sortedWords = wlist.map(lambda x: (''.join(sorted(x)), set([x])))
+  mergedWords = sortedWords.reduceByKey(lambda s1, s2: s1 | s2)
+  finalRdd = mergedWords.map(lambda x: (x[0], len(x[1]),list(x[1])))
+  finalRdd.cache()
+
+  maxlen = 0
+  anagram = None
+  maxItems = finalRdd.takeOrdered(20, lambda x: -x[1])
+  # print maxItems
+  for item in maxItems:
```

Comment:
Unclear why do you need this for loop. Instead you can change the value of takeOrdered to 1 (instead of 20).